src/auspex/instruments/prologix.py

```diff
@@ -0,0 +1,292 @@
+        """Write a string message to device in ASCII format.
+
+        Args:
+            command: The message to be sent.
+        Returns:
+            The number of bytes in the message.
+        """
+        self._addr()
+        self.sock.send((command + self.write_termination).encode())
+        return len(command)
+
+    def read_raw(self, bufsize=None):
+        """Read bytes from instrument.
+
+        Args:
+            bufsize: Number of bytes to read from instrument. Defaults to resource
+            bufsize if None.
+        Returns:
+            Instrument data. Nothing is stripped from response.
+        """
+        if bufsize is None:
+            bufsize = self.bufsize
+        self._addr()
+        return self.socket.recv(bufsize)
+
+    def write_raw(self, command):
+        """Write a string message to device as raw bytes. No termination
+        character is appended.
+
+        Args:
+            command: The message to be sent.
+        Returns:
+            The number of bytes in the message.
+        """
+        self._addr()
+        self.sock.send(command)
+        return len(command)
+
+    def write_ascii_values(self, command, values, converter='f', separator=','):
+        """Write a string message to device followed by values in ASCII format.
+
+        Args:
+            command: Message to be sent to device.
+            values: Data to be written to device (as an iterable)
+            converter: String format code to be used to convert values.
+            separator: Separator between values -- separator.join(data).
+
+        Returns:
+            Total number of bytes sent to instrument.
+        """
+        converter = '%' + converter
+        ascii_vals = separator.join(converter % v for v in values)
+        return self.write(command + ascii_vals + self.write_termination)
+
+    def query_ascii_values(self, command, converter='f', separator=',',
+        container=list, bufsize=None):
+        """Write a string message to device and return values as iterable.
+
+        Args:
+            command: Message to be sent to device.
+            values: Data to be written to device (as an interable)
+            converter: String format code to be used to convert values.
+            separator: Separator between values -- data.split(separator).
+            container: Iterable type to use for output.
+            bufsize: Number of bytes to read from instrument. Defaults to resource
+            bufsize if None.
+        Returns:
+            Iterable of values converted from instrument response.
+        """
+        try:
+            converter = _converters[converter]
+        except KeyError as err:
+            raise ValueError("Invalid string converter: {0}".format(converter)) from err
+
+        ascii = self.query(command, bufsize=bufsize)
+        return container([converter(v) for v in ascii.split(separator)])
+
+    def write_binary_values(self, command, values, datatype='f',
```

Comment:
is it more fragile to reimplement this ourselves or import it from `pyvisa.util`?